src/storage/ExpiringLock.ts

```diff
@@ -0,0 +1,18 @@
+import type { EventEmitter } from 'events';
+import type { Lock } from './Lock';
+
+/**
+ * ExpiringLock used by a {@link ExpiringResourceLocker} for non-atomic operations.
+ * Emits an "expired" event when internal timer runs out and should call release function when this happen.
+ */
+export interface ExpiringLock extends Lock, EventEmitter {
+  /**
+   * Time in ms after which reading a representation times out, causing the lock to be released.
+   */
+  readTimeout: number;
```

Comment:
No need to expose this.

Reply:
Hmm but then I'll not be able to print the timeout in the [log statements](https://github.com/solid/community-server/pull/286/files#diff-c5cbc6d4338adb4b7ff1a21e5bc61dc918e7d2baba5dccdf47ec2328a43c7217R124-R125):
```typescript
this.logger.info(`Stream reading timout of ${lock.readTimeout}ms exceeded; destroying source`);
source.destroy(new Error(`Stream reading timout of ${lock.readTimeout}ms exceeded`));
```
I could:
a. Not specify the specific amount of ms in the sentences. 
b. Add readTimeout number to `ExpiringResourceLocker` interface (what just expose it in another place)
c. Pass `readTimeout` as a constructor parameter of `LockingResourceStore`

Reply:
d. Have the lock implementation print this message itself 😉

It knows its own timeout, but doesn't have to expose it.

Reply:
Yes, that's possible for the logger statement, but not for the parameter of `new Error`, but guess I'll just add amount of ms to the already existing logger.verbose in the lock implementation and change the second line to `source.destroy(new Error(`Stream reading timout exceeded`));`
Agreed?

Reply:
Sounds good to me, thanks!